src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,118 @@
+#include "EndpointQueueFilter.h"
+#include <algorithm>
+#include <cctype>
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+
+namespace {
+
+bool IsValidMdnsHostName(const CharSpan & hostName)
+{
+    for (size_t i = 0; i < hostName.size(); ++i)
+    {
+        char ch_data = *(hostName.data());
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F') || (ch_data >= 'a' && ch_data <= 'f')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool IsMdnsBroadcastPacket(const IPPacketInfo & pktInfo, const System::PacketBufferHandle & pktPayload)
+{
+    // if the packet is not a broadcast packet to mDNS port, drop it.
+    VerifyOrReturnValue(pktInfo.DestPort == 5353, false);
+#if INET_CONFIG_ENABLE_IPV4
+    ip_addr_t mdnsIPv4BroadcastAddr = IPADDR4_INIT_BYTES(224, 0, 0, 251);
+    if (pktInfo.DestAddress == Inet::IPAddress(mdnsIPv4BroadcastAddr)
+    {
+        return true;
+    }
+#endif
+    ip_addr_t mdnsIPv6BroadcastAddr = IPADDR6_INIT_HOST(0xFF020000, 0, 0, 0xFB);
+    if (pktInfo.DestAddress == Inet::IPAddress(mdnsIPv6BroadcastAddr))
+    {
+        return true;
+    }
+    return false;
+}
+
+bool PayloadContainsCaseInsensitive(const System::PacketBufferHandle & payload, const ByteSpan & pattern)
+{
+    if (payload->TotalLength() == 0 || pattern.size() == 0)
+    {
+        return false;
+    }
+
+    if (payload->HasChainedBuffer() || payload->TotalLength() < pattern.size())
+    {
+        return false;
+    }
+
+    CharSpan payloadView(reinterpret_cast<const char *>(payload->Start()), payload->TotalLength());
+    CharSpan patternView(reinterpret_cast<const char *>(pattern.data()), pattern.size());
+
+    auto toLower = [](char c) { return std::tolower(static_cast<unsigned char>(c)); };
+
+    auto it = std::search(payloadView.data(), payloadView.data() + payloadView.size(), patternView.data(),
+                          patternView.data() + patternView.size(), [&](char a, char b) { return toLower(a) == toLower(b); });
```

Comment:
So I have to ask: if we are going to just be passing these to something that wants unsigned char, is there a reason we are not just using `Span<unsigned char>` instead of `CharSpan`?  And for that matter, having pattern be a `Span<unsigned char>` to start with?

Reply:
changed the CharSpan to Span<unsigned char>